src/main/java/org/jenkinsci/plugins/bigpanda/ChangesBuilder.java

```diff
@@ -15,49 +19,115 @@
-        json.put("ticket_url", Jenkins.getInstance().getRootUrl() + this.build.getUrl());
+        this.json.put("identifier", String.valueOf(this.build.getDisplayName()));
+        this.json.put("start", (this.build.getStartTimeInMillis() / 1000));
+        this.json.put("summary", this.build.getFullDisplayName().replaceAll("[^a-zA-Z0-9-# ]", ""));
+        this.json.put("ticket_url", Jenkins.getInstance().getRootUrl() + this.build.getUrl());
 
-        if(getEndTime() != 0) {
-            json.put("end", getEndTime());
+        if (getEndTime() != 0) {
+            this.json.put("end", getEndTime());
         }
+    }
 
-        // Adding remaining fields to tags object
-        JSONObject tags = new JSONObject();
-        tags.put("description", getBuildDescription());
+    private void addUserInfo() {
 
         List<Cause> causes = this.build.getCauses();
-        String[] causesArray = new String[causes.size()];
-        for (int i = 0; i < causes.size(); i++) {
-            causesArray[i] = causes.get(i).getShortDescription();
-        }
 
-        tags.put("causes", causesArray);
+        for (Cause cause : causes) {
+            if (cause instanceof Cause.UserIdCause) {
+                String userId = ((Cause.UserIdCause) cause).getUserId();
+                userId = (userId == null || userId.isEmpty()) ? "Anonymous" : userId;
+
+                this.tags.put("user_id", userId);
+
+                String userName = ((Cause.UserIdCause) cause).getUserName();
+                userName = (userName == null || userName.isEmpty() ? "Anonymous" : userName);
+                
+                this.tags.put("user_name", userName);
+                break;
+
+            } else if (cause instanceof Cause.UpstreamCause) {
+                this.tags.put("user_id", "upstream");
+                break;
+            } else if (cause instanceof TimerTrigger.TimerTriggerCause) {
+                this.tags.put("user_id", "timer");
+                break;
+            } else {
+                this.tags.put("user_id", "Unknown");
+                this.tags.put("user_name", "SYSTEM");
+                break;
+            }
+        }
+    }
 
-        // Add the tags object
-        json.put("tags", tags);
+    /**
+     * Get the SCM info from the build and add it to tags.
+     *
+     */
+    private void addSCMInfo() {
+        EnvVars environment = null;
+        try {
+            environment = this.build.getEnvironment(this.listener);
+
+            this.listener.getLogger().println(environment);
+        } catch (IOException e) {
+            this.listener.getLogger().println("BigPanda Notifier: Could not retrieve environment.");
+        } catch (InterruptedException e) {
+            this.listener.getLogger().println("BigPanda Notifier: Could not retrieve environment.");
+        }
 
-        return json;
+        if (environment != null) {
+            if (environment.get("GIT_URL") != null) {
```

Comment:
IIRC, you have `NODE_NAME` and `NODE_LABELS` that are worth adding here as well.